tests/ifs/ifs.c

```diff
@@ -206,9 +206,12 @@ static int scan_run(struct test *test, int cpu)
         char result[BUFLEN] = {}, my_cpu[BUFLEN] = {};
         unsigned long long code;
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-variable"
         /* HACK: Shadows global variable that log_warning() uses
          * DON'T use report_fail_msg() */
         int thread_num = cpu;
+#pragma GCC diagnostic pop
```

Comment:
But then why is the compiler complaining? I don't see a `log_warning` in this function, but there are `log_error` and  `log_skip`. Was this in the `NO_LOGGING` build?
https://github.com/opendcdiag/opendcdiag/blob/b194024e529c06b15f8754daee828f97a1ed7130/framework/sandstone.h#L780-L783

Reply:
That warning is seen with `debug`, `debugoptimized`, and `release` configurations. Here's the output from the debug build (unittests target):

```
meson setup --buildtype debug build-debug .

ninja -C build-debug unittests
ninja: Entering directory `build-debug'
[11/16] Compiling C++ object unittests.p/tests_ifs_unit_ifs_unittests.cpp.o
In file included from ../tests/ifs/unit/ifs_unittests.cpp:15:
../tests/ifs/unit/../ifs.c: In function ‘int scan_run(test*, int)’:
../tests/ifs/unit/../ifs.c:211:13: warning: unused variable ‘thread_num’ [-Wunused-variable]
  211 |         int thread_num = cpu;
      |             ^~~~~~~~~~
      
< ... snip ... >

[16/16] Linking target unittests
```

Reply:
Just add a `(void) thread_num;` to silence the warning. No need for pragmas.